tests/schedule_test.go

```diff
@@ -1186,30 +1138,29 @@ func (s *ScheduleFunctionalSuite) TestNextTimeCache() {
 	)
 	s.Equal(expectedRefills+uuidCacheRefills, sideEffects)
 	s.Equal(expectedRefills, nextTimeSideEffects)
-
-	// cleanup
-	_, err = s.FrontendClient().DeleteSchedule(testcore.NewContext(), &workflowservice.DeleteScheduleRequest{
-		Namespace:  s.Namespace(),
-		ScheduleId: sid,
-		Identity:   "test",
-	})
-	s.NoError(err)
 }
 
-func (s *ScheduleFunctionalSuite) getScheduleEntryFomVisibility(sid string) *schedulepb.ScheduleListEntry {
+func (s *ScheduleFunctionalSuite) getScheduleEntryFomVisibility(sid string, requiredRecentActions int) *schedulepb.ScheduleListEntry {
 	var slEntry *schedulepb.ScheduleListEntry
-	s.Eventually(func() bool { // wait for visibility
+	s.Require().Eventually(func() bool { // wait for visibility
 		listResp, err := s.FrontendClient().ListSchedules(testcore.NewContext(), &workflowservice.ListSchedulesRequest{
 			Namespace:       s.Namespace(),
 			MaximumPageSize: 5,
 		})
-		if err != nil || len(listResp.Schedules) != 1 || listResp.Schedules[0].ScheduleId != sid ||
-			len(listResp.Schedules[0].GetInfo().GetRecentActions()) < 2 {
+		if err != nil {
 			return false
 		}
-		slEntry = listResp.Schedules[0]
-		return true
-	}, 10*time.Second, 1*time.Second)
+		for _, ent := range listResp.Schedules {
+			if ent.ScheduleId == sid {
+				if len(ent.GetInfo().GetRecentActions()) < requiredRecentActions {
+					return false
+				}
```

Comment:
In a test I'm writing for `ListSchedules`, I test against having `RunningWorkflows`; what do you think of having this helper take a predicate instead of `requiredRecentActions`? (Happy to change it as part of my PR, as well).

Reply:
oh, yeah I like that